
.. DO NOT EDIT.
.. THIS FILE WAS AUTOMATICALLY GENERATED BY SPHINX-GALLERY.
.. TO MAKE CHANGES, EDIT THE SOURCE PYTHON FILE:
.. "tutorials/DataCollectionHDF5.py"
.. LINE NUMBERS ARE GIVEN BELOW.

.. only:: html

    .. note::
        :class: sphx-glr-download-link-note

        Click :ref:`here <sphx_glr_download_tutorials_DataCollectionHDF5.py>`
        to download the full example code

.. rst-class:: sphx-glr-example-title

.. _sphx_glr_tutorials_DataCollectionHDF5.py:


HDF5 Database
=============

This example is nearly identical to the MSEED tutorial except the fact that we
are specifying a different data format in the request dictionary.

Downloading Event & Station Metadata
------------------------------------

In this section, the only difference is the ``format`` ``key`` in the
``request_dict`` that is set to ``hdf5``.

.. GENERATED FROM PYTHON SOURCE LINES 16-19

.. code-block:: default

    # sphinx_gallery_thumbnail_number = 1
    # sphinx_gallery_dummy_images = 1








.. GENERATED FROM PYTHON SOURCE LINES 20-21

First let's get a path where to create the data.

.. GENERATED FROM PYTHON SOURCE LINES 21-71

.. code-block:: default



    # Some needed Imports
    import os
    from typing import NewType
    from obspy import UTCDateTime
    from pyglimer.waveform.request import Request

    # Get notebook path for future reference of the database:
    # Get notebook path for future reference of the database:
    try: db_base_path = ipynb_path
    except NameError:
        try: db_base_path = os.path.dirname(os.path.realpath(__file__))
        except NameError: db_base_path = os.getcwd()

    # Define file locations
    proj_dir = os.path.join(db_base_path, 'tmp', 'database_hdf5')

    # Define network and station to download RFs for
    network = 'IU'
    station = 'HRV'

    request_dict = {
        # Necessary arguments
        'proj_dir': proj_dir,
        'raw_subdir': os.path.join('waveforms', 'raw'),# Directory of the waveforms
        'prepro_subdir': os.path.join('waveforms', 'preprocessed'),  # Directory of the preprocessed waveforms
        'rf_subdir': os.path.join('waveforms', 'RF'),  # Directory of the receiver functions
        'statloc_subdir': 'stations', # Directory stations
        'evt_subdir': 'events',       # Directory of the events
        'log_subdir': 'log',          # Directory for the logs
        'loglvl': 'WARNING',          # logging level
        'format': 'hdf5',              # Format to save database in
        "phase": "P",                 # 'P' or 'S' receiver functions
        "rot": "RTZ",                 # Coordinate system to rotate to
        "deconmeth": "waterlevel",    # Deconvolution method
        "starttime": UTCDateTime(2021, 1, 1, 0, 0, 0), # Starttime of database.
                                                    # Here, starttime of HRV
        "endtime": UTCDateTime(2021, 7, 1, 0, 0, 0), # Endtimetime of database
        # kwargs below
        "pol": 'v',                   # Source wavelet polaristion. Def. "v" --> SV
        "minmag": 5.5,                # Earthquake minimum magnitude. Def. 5.5
        "event_coords": None,         # Specific event?. Def. None
        "network": network,              # Restricts networks. Def. None
        "station": station,             # Restricts stations. Def. None
        "waveform_client": ["IRIS"],  # FDSN server client (s. obspy). Def. None
        "evtcat": None,               # If you have already downloaded a set of
                                    # events previously, you can use them here
    }








.. GENERATED FROM PYTHON SOURCE LINES 72-74

Now that all parameters are in place, let's initialize the 
:class:`pyglimer.waveform.request.Request`

.. GENERATED FROM PYTHON SOURCE LINES 74-78

.. code-block:: default


    # Initializing the Request class and downloading the data
    R = Request(**request_dict)








.. GENERATED FROM PYTHON SOURCE LINES 79-81

The initialization will look for all events for which data is available. To see 
whether the events make sense we plot a map of the events:

.. GENERATED FROM PYTHON SOURCE LINES 81-92

.. code-block:: default


    import matplotlib.pyplot as plt
    from pyglimer.plot.plot_utils import plot_catalog
    from pyglimer.plot.plot_utils import set_mpl_params

    # Setting plotting parameters
    set_mpl_params()

    # Plotting the catalog
    plot_catalog(R.evtcat)




.. image-sg:: /tutorials/images/sphx_glr_DataCollectionHDF5_001.png
   :alt: DataCollectionHDF5
   :srcset: /tutorials/images/sphx_glr_DataCollectionHDF5_001.png
   :class: sphx-glr-single-img





.. GENERATED FROM PYTHON SOURCE LINES 93-94

We can also quickly check how many events we gathered.

.. GENERATED FROM PYTHON SOURCE LINES 94-97

.. code-block:: default


    print(f"There are {len(R.evtcat)} available events")





.. rst-class:: sphx-glr-script-out

 Out:

 .. code-block:: none

    There are 289 available events




.. GENERATED FROM PYTHON SOURCE LINES 98-102

-------------------------------------------------

Again, this section does not really change, because the ``request_dict``
parsed all needed information to :class:`pyglimer.waveform.request.Request`.

.. GENERATED FROM PYTHON SOURCE LINES 102-105

.. code-block:: default


    R.download_waveforms_small_db(channel='BH?')





.. rst-class:: sphx-glr-script-out

 Out:

 .. code-block:: none

    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-06-26T18:08:17__2021-06-26T18:10:47__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-06-26T18:08:17__2021-06-26T18:10:47__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-06-26T18:08:17__2021-06-26T18:10:47__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-06-23T03:03:16__2021-06-23T03:05:46__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-06-23T03:03:16__2021-06-23T03:05:46__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-06-23T03:03:16__2021-06-23T03:05:46__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-06-21T22:25:07__2021-06-21T22:27:37__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-06-21T22:25:07__2021-06-21T22:27:37__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-06-21T22:25:07__2021-06-21T22:27:37__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-06-13T03:04:16__2021-06-13T03:06:46__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-06-13T03:04:16__2021-06-13T03:06:46__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-06-13T03:04:16__2021-06-13T03:06:46__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-06-10T23:33:29__2021-06-10T23:35:59__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-06-10T23:33:29__2021-06-10T23:35:59__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-06-10T23:33:29__2021-06-10T23:35:59__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-06-04T08:24:02__2021-06-04T08:26:32__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-06-04T08:24:02__2021-06-04T08:26:32__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-06-04T08:24:02__2021-06-04T08:26:32__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-06-04T07:59:45__2021-06-04T08:02:15__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-06-04T07:59:45__2021-06-04T08:02:15__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-06-04T07:59:45__2021-06-04T08:02:15__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-06-01T11:27:49__2021-06-01T11:30:19__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-06-01T11:27:49__2021-06-01T11:30:19__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-06-01T11:27:49__2021-06-01T11:30:19__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-05-31T07:07:55__2021-05-31T07:10:25__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-05-31T07:07:55__2021-05-31T07:10:25__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-05-31T07:07:55__2021-05-31T07:10:25__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-05-29T01:15:36__2021-05-29T01:18:06__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-05-29T01:15:36__2021-05-29T01:18:06__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-05-29T01:15:36__2021-05-29T01:18:06__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-05-21T22:25:51__2021-05-21T22:28:21__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-05-21T22:25:51__2021-05-21T22:28:21__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-05-21T22:25:51__2021-05-21T22:28:21__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-05-19T18:06:49__2021-05-19T18:09:19__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-05-19T18:06:49__2021-05-19T18:09:19__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-05-19T18:06:49__2021-05-19T18:09:19__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-05-19T00:54:14__2021-05-19T00:56:44__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-05-19T00:54:14__2021-05-19T00:56:44__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-05-19T00:54:14__2021-05-19T00:56:44__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-05-16T03:36:21__2021-05-16T03:38:51__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-05-16T03:36:21__2021-05-16T03:38:51__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-05-16T03:36:21__2021-05-16T03:38:51__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-05-14T00:11:00__2021-05-14T00:13:30__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-05-14T00:11:00__2021-05-14T00:13:30__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-05-14T00:11:00__2021-05-14T00:13:30__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-05-13T09:48:51__2021-05-13T09:51:20__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-05-13T09:48:51__2021-05-13T09:51:20__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-05-13T09:48:51__2021-05-13T09:51:20__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-05-12T16:06:05__2021-05-12T16:08:35__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-05-12T16:06:05__2021-05-12T16:08:35__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-05-12T16:06:05__2021-05-12T16:08:35__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-05-12T08:51:51__2021-05-12T08:54:21__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-05-12T08:51:51__2021-05-12T08:54:21__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-05-12T08:51:51__2021-05-12T08:54:21__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-05-12T06:23:13__2021-05-12T06:25:43__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-05-12T06:23:13__2021-05-12T06:25:43__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-05-12T06:23:13__2021-05-12T06:25:43__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-05-07T11:51:44__2021-05-07T11:54:14__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-05-07T11:51:44__2021-05-07T11:54:14__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-05-07T11:51:44__2021-05-07T11:54:14__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-05-03T08:58:49__2021-05-03T09:01:19__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-05-03T08:58:49__2021-05-03T09:01:19__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-05-03T08:58:49__2021-05-03T09:01:19__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-05-02T07:08:31__2021-05-02T07:11:01__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-05-02T07:08:31__2021-05-02T07:11:01__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-05-02T07:08:31__2021-05-02T07:11:01__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-05-01T01:40:10__2021-05-01T01:42:40__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-05-01T01:40:10__2021-05-01T01:42:40__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-05-01T01:40:10__2021-05-01T01:42:40__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-04-29T02:28:37__2021-04-29T02:31:07__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-04-29T02:28:37__2021-04-29T02:31:07__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-04-29T02:28:37__2021-04-29T02:31:07__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-04-25T23:59:52__2021-04-26T00:02:22__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-04-25T23:59:52__2021-04-26T00:02:22__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-04-25T23:59:52__2021-04-26T00:02:22__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-04-22T00:28:33__2021-04-22T00:31:03__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-04-22T00:28:33__2021-04-22T00:31:03__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-04-22T00:28:33__2021-04-22T00:31:03__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-04-18T06:54:22__2021-04-18T06:56:51__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-04-18T06:54:22__2021-04-18T06:56:51__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-04-18T06:54:22__2021-04-18T06:56:51__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-04-18T00:42:09__2021-04-18T00:44:39__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-04-18T00:42:09__2021-04-18T00:44:39__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-04-18T00:42:09__2021-04-18T00:44:39__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-04-18T00:41:53__2021-04-18T00:44:23__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-04-18T00:41:53__2021-04-18T00:44:23__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-04-18T00:41:53__2021-04-18T00:44:23__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-04-14T04:41:40__2021-04-14T04:44:10__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-04-14T04:41:40__2021-04-14T04:44:10__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-04-14T04:41:40__2021-04-14T04:44:10__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-04-08T17:18:15__2021-04-08T17:20:45__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-04-08T17:18:15__2021-04-08T17:20:45__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-04-08T17:18:15__2021-04-08T17:20:45__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-03-30T21:47:47__2021-03-30T21:50:17__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-03-30T21:47:47__2021-03-30T21:50:17__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-03-30T21:47:47__2021-03-30T21:50:17__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-03-28T23:07:33__2021-03-28T23:10:03__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-03-28T23:07:33__2021-03-28T23:10:03__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-03-28T23:07:33__2021-03-28T23:10:03__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-03-27T13:57:43__2021-03-27T14:00:12__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-03-27T13:57:43__2021-03-27T14:00:12__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-03-27T13:57:43__2021-03-27T14:00:12__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-03-20T09:22:26__2021-03-20T09:24:56__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-03-20T09:22:26__2021-03-20T09:24:56__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-03-20T09:22:26__2021-03-20T09:24:56__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-03-20T03:12:51__2021-03-20T03:15:20__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-03-20T03:12:51__2021-03-20T03:15:20__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-03-20T03:12:51__2021-03-20T03:15:20__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-03-20T01:55:09__2021-03-20T01:57:39__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-03-20T01:55:09__2021-03-20T01:57:39__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-03-20T01:55:09__2021-03-20T01:57:39__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-03-18T00:13:25__2021-03-18T00:15:55__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-03-18T00:13:25__2021-03-18T00:15:55__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-03-18T00:13:25__2021-03-18T00:15:55__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-03-17T00:24:49__2021-03-17T00:27:19__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-03-17T00:24:49__2021-03-17T00:27:19__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-03-17T00:24:49__2021-03-17T00:27:19__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-03-16T18:49:16__2021-03-16T18:51:46__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-03-16T18:49:16__2021-03-16T18:51:46__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-03-16T18:49:16__2021-03-16T18:51:46__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-03-12T13:08:13__2021-03-12T13:10:43__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-03-12T13:08:13__2021-03-12T13:10:43__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-03-12T13:08:13__2021-03-12T13:10:43__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-03-04T19:34:13__2021-03-04T19:36:42__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-03-04T19:34:13__2021-03-04T19:36:42__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-03-04T19:34:13__2021-03-04T19:36:42__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-03-04T18:48:41__2021-03-04T18:51:11__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-03-04T18:48:41__2021-03-04T18:51:11__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-03-04T18:48:41__2021-03-04T18:51:11__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-03-03T10:26:32__2021-03-03T10:29:02__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-03-03T10:26:32__2021-03-03T10:29:02__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-03-03T10:26:32__2021-03-03T10:29:02__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-03-02T21:34:57__2021-03-02T21:37:27__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-03-02T21:34:57__2021-03-02T21:37:27__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-03-02T21:34:57__2021-03-02T21:37:27__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-02-28T02:25:39__2021-02-28T02:28:09__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-02-28T02:25:39__2021-02-28T02:28:09__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-02-28T02:25:39__2021-02-28T02:28:09__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-02-24T10:12:23__2021-02-24T10:14:52__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-02-24T10:12:23__2021-02-24T10:14:52__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-02-24T10:12:23__2021-02-24T10:14:52__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-02-18T19:41:18__2021-02-18T19:43:48__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-02-18T19:41:18__2021-02-18T19:43:48__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-02-18T19:41:18__2021-02-18T19:43:48__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-02-17T03:46:34__2021-02-17T03:49:04__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-02-17T03:46:34__2021-02-17T03:49:04__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-02-17T03:46:34__2021-02-17T03:49:04__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-02-13T14:20:35__2021-02-13T14:23:04__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-02-13T14:20:35__2021-02-13T14:23:04__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-02-13T14:20:35__2021-02-13T14:23:04__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-02-12T17:14:10__2021-02-12T17:16:40__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-02-12T17:14:10__2021-02-12T17:16:40__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-02-12T17:14:10__2021-02-12T17:16:40__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-02-10T08:32:54__2021-02-10T08:35:24__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-02-10T08:32:54__2021-02-10T08:35:24__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-02-10T08:32:54__2021-02-10T08:35:24__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-02-03T05:35:33__2021-02-03T05:38:03__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-02-03T05:35:33__2021-02-03T05:38:03__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-02-03T05:35:33__2021-02-03T05:38:03__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-01-31T19:12:28__2021-01-31T19:14:58__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-01-31T19:12:28__2021-01-31T19:14:58__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-01-31T19:12:28__2021-01-31T19:14:58__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-01-29T13:21:12__2021-01-29T13:23:42__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-01-29T13:21:12__2021-01-29T13:23:42__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-01-29T13:21:12__2021-01-29T13:23:42__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-01-29T12:48:06__2021-01-29T12:50:36__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-01-29T12:48:06__2021-01-29T12:50:36__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-01-29T12:48:06__2021-01-29T12:50:36__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-01-24T00:18:47__2021-01-24T00:21:17__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-01-24T00:18:47__2021-01-24T00:21:17__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-01-24T00:18:47__2021-01-24T00:21:17__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-01-20T23:08:30__2021-01-20T23:11:00__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-01-20T23:08:30__2021-01-20T23:11:00__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-01-20T23:08:30__2021-01-20T23:11:00__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-01-20T16:11:30__2021-01-20T16:13:59__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-01-20T16:11:30__2021-01-20T16:13:59__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-01-20T16:11:30__2021-01-20T16:13:59__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-01-19T02:57:26__2021-01-19T02:59:56__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-01-19T02:57:26__2021-01-19T02:59:56__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-01-19T02:57:26__2021-01-19T02:59:56__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-01-16T20:15:11__2021-01-16T20:17:41__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-01-16T20:15:11__2021-01-16T20:17:41__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-01-16T20:15:11__2021-01-16T20:17:41__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-01-15T21:43:58__2021-01-15T21:46:28__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-01-15T21:43:58__2021-01-15T21:46:28__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-01-15T21:43:58__2021-01-15T21:46:28__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-01-15T16:17:00__2021-01-15T16:19:30__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-01-15T16:17:00__2021-01-15T16:19:30__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-01-15T16:17:00__2021-01-15T16:19:30__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-01-15T13:31:57__2021-01-15T13:34:27__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-01-15T13:31:57__2021-01-15T13:34:27__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-01-15T13:31:57__2021-01-15T13:34:27__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-01-12T02:51:46__2021-01-12T02:54:16__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-01-12T02:51:46__2021-01-12T02:54:16__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-01-12T02:51:46__2021-01-12T02:54:16__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-01-11T21:56:15__2021-01-11T21:58:45__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-01-11T21:56:15__2021-01-11T21:58:45__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-01-11T21:56:15__2021-01-11T21:58:45__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-01-11T21:53:15__2021-01-11T21:55:44__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-01-11T21:53:15__2021-01-11T21:55:44__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-01-11T21:53:15__2021-01-11T21:55:44__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-01-11T21:45:10__2021-01-11T21:47:40__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-01-11T21:45:10__2021-01-11T21:47:40__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-01-11T21:45:10__2021-01-11T21:47:40__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-01-10T15:56:47__2021-01-10T15:59:17__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-01-10T15:56:47__2021-01-10T15:59:17__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-01-10T15:56:47__2021-01-10T15:59:17__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-01-10T04:04:10__2021-01-10T04:06:40__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-01-10T04:04:10__2021-01-10T04:06:40__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-01-10T04:04:10__2021-01-10T04:06:40__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH1__2021-01-03T12:49:14__2021-01-03T12:51:44__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BH2__2021-01-03T12:49:14__2021-01-03T12:51:44__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)
    /usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/pyasdf/asdf_data_set.py:1546: ASDFWarning: Data 'IU.HRV/IU.HRV.60.BHZ__2021-01-03T12:49:14__2021-01-03T12:51:44__raw_recording' already exists in file. Will not be added!
      warnings.warn(msg, ASDFWarning)




.. GENERATED FROM PYTHON SOURCE LINES 106-109

there is indeed a change in the code because we saved the raw data in form of
``ASDFDataset``s and we need to actually access the ``hdf5`` file that we
to count how many traces it contains.

.. GENERATED FROM PYTHON SOURCE LINES 109-134

.. code-block:: default


    # Import ASDFDataset
    from pyasdf import ASDFDataSet

    # Path to the where the miniseeds are stored
    data_storage = os.path.join(
        proj_dir, 'waveforms', 'raw', 'P', f'{network}.{station}.h5')

    # Read the data from the station ``h5`` file
    with ASDFDataSet(data_storage, mode='r', mpi=False) as ds:

        # Perform waveform query on ASDFDataSet
        stream = ds.get_waveforms(
            network,
            station,
            '*', # Location
            '*', # Channel
            request_dict['starttime']-1000,  # Starttime 
            request_dict['endtime']+1000,    # Endtime
            'raw_recording')

    # Print output
    print(f"Number of downloaded waveforms: {len(stream)}")






.. rst-class:: sphx-glr-script-out

 Out:

 .. code-block:: none

    Number of downloaded waveforms: 213




.. GENERATED FROM PYTHON SOURCE LINES 135-160

The final step to get you receiver function data is the preprocessing. 
Although it is hidden in a single function, which
is :func:`pyglimer.waveform.request.Request.preprocess`
A lot of decisions are being made:

Processing steps:
1. Clips waveform to the right length (tz before and ta after theorethical 
arrival.)
2. Demean & Detrend
3. Tapering
4. Remove Instrument response, convert to velocity &
simulate havard station
5. Rotation to NEZ and, subsequently, to RTZ.
6. Compute SNR for highpass filtered waveforms (highpass f defined in 
qc.lowco) If SNR lower than in qc.SNR_criteria for all filters, rejects 
waveform.
7. Write finished and filtered waveforms to folder
specified in qc.outputloc.
8. Write info file with shelf containing station,
event and waveform information.
9. (Optional) If we had chosen a different coordinate system in ``rot``
than RTZ, it would now cast the preprocessed waveforms information
that very coordinate system.
10. Deconvolution with method ``deconmeth`` from our dict is perfomed.


.. GENERATED FROM PYTHON SOURCE LINES 160-163

.. code-block:: default


    R.preprocess(hc_filt=1.5, client='single')





.. rst-class:: sphx-glr-script-out

 Out:

 .. code-block:: none

      0%|          | 0/1 [00:00<?, ?it/s]
      0%|          | 0/289 [00:00<?, ?it/s]
      0%|          | 1/289 [00:00<00:46,  6.17it/s]
      1%|1         | 4/289 [00:00<00:21, 13.09it/s]
      2%|2         | 6/289 [00:00<00:22, 12.82it/s]
      6%|5         | 17/289 [00:00<00:08, 33.86it/s]
      7%|7         | 21/289 [00:00<00:08, 30.60it/s]
      9%|8         | 25/289 [00:01<00:12, 21.80it/s]
     10%|9         | 28/289 [00:01<00:15, 16.35it/s]
     10%|#         | 30/289 [00:01<00:17, 15.13it/s]
     11%|#1        | 32/289 [00:01<00:17, 14.35it/s]
     14%|#4        | 41/289 [00:01<00:10, 24.75it/s]
     15%|#5        | 44/289 [00:02<00:13, 17.57it/s]
     16%|#6        | 47/289 [00:02<00:13, 17.92it/s]
     17%|#7        | 50/289 [00:02<00:13, 18.16it/s]
     18%|#8        | 53/289 [00:02<00:16, 14.48it/s]
     19%|#9        | 55/289 [00:03<00:16, 14.00it/s]
     20%|##        | 59/289 [00:03<00:13, 16.57it/s]
     21%|##1       | 62/289 [00:03<00:13, 17.07it/s]
     22%|##2       | 65/289 [00:03<00:13, 16.75it/s]
     23%|##3       | 67/289 [00:03<00:14, 15.62it/s]
     24%|##4       | 70/289 [00:03<00:13, 16.27it/s]
     27%|##6       | 77/289 [00:04<00:08, 23.88it/s]
     28%|##8       | 81/289 [00:04<00:08, 24.12it/s]
     29%|##9       | 85/289 [00:04<00:08, 23.77it/s]2022-06-24 12:34:06,257 - ERROR - RF Creation failed. Waveform Data:
    IU.HRV.2021108T002925.0380Z
    original error:
    Unexpected behavior in rotation. Please file a bug report on github.
    Traceback (most recent call last):
      File "/home/runner/work/PyGLImER/PyGLImER/src/pyglimer/waveform/preprocessh5.py", line 346, in __station_process__
        st, _, infodict = __rotate_qc(
      File "/home/runner/work/PyGLImER/PyGLImER/src/pyglimer/waveform/preprocessh5.py", line 405, in __rotate_qc
        st.rotate(method='->ZNE', inventory=station_inv)
      File "/usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/obspy/core/stream.py", line 2733, in rotate
        return self._rotate_to_zne(inventory, **kwargs)
      File "/usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/obspy/core/stream.py", line 3469, in _rotate_to_zne
        self._rotate_specific_channels_to_zne(
      File "/usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/obspy/core/stream.py", line 3542, in _rotate_specific_channels_to_zne
        raise NotImplementedError(msg)
    NotImplementedError: Unexpected behavior in rotation. Please file a bug report on github.
    ERROR:pyglimer.request.preprocess.RF:RF Creation failed. Waveform Data:
    IU.HRV.2021108T002925.0380Z
    original error:
    Unexpected behavior in rotation. Please file a bug report on github.
    Traceback (most recent call last):
      File "/home/runner/work/PyGLImER/PyGLImER/src/pyglimer/waveform/preprocessh5.py", line 346, in __station_process__
        st, _, infodict = __rotate_qc(
      File "/home/runner/work/PyGLImER/PyGLImER/src/pyglimer/waveform/preprocessh5.py", line 405, in __rotate_qc
        st.rotate(method='->ZNE', inventory=station_inv)
      File "/usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/obspy/core/stream.py", line 2733, in rotate
        return self._rotate_to_zne(inventory, **kwargs)
      File "/usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/obspy/core/stream.py", line 3469, in _rotate_to_zne
        self._rotate_specific_channels_to_zne(
      File "/usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/obspy/core/stream.py", line 3542, in _rotate_specific_channels_to_zne
        raise NotImplementedError(msg)
    NotImplementedError: Unexpected behavior in rotation. Please file a bug report on github.
    2022-06-24 12:34:06,504 - ERROR - RF Creation failed. Waveform Data:
    IU.HRV.2021108T002912.4990Z
    original error:
    Unexpected behavior in rotation. Please file a bug report on github.
    Traceback (most recent call last):
      File "/home/runner/work/PyGLImER/PyGLImER/src/pyglimer/waveform/preprocessh5.py", line 346, in __station_process__
        st, _, infodict = __rotate_qc(
      File "/home/runner/work/PyGLImER/PyGLImER/src/pyglimer/waveform/preprocessh5.py", line 405, in __rotate_qc
        st.rotate(method='->ZNE', inventory=station_inv)
      File "/usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/obspy/core/stream.py", line 2733, in rotate
        return self._rotate_to_zne(inventory, **kwargs)
      File "/usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/obspy/core/stream.py", line 3469, in _rotate_to_zne
        self._rotate_specific_channels_to_zne(
      File "/usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/obspy/core/stream.py", line 3542, in _rotate_specific_channels_to_zne
        raise NotImplementedError(msg)
    NotImplementedError: Unexpected behavior in rotation. Please file a bug report on github.
    ERROR:pyglimer.request.preprocess.RF:RF Creation failed. Waveform Data:
    IU.HRV.2021108T002912.4990Z
    original error:
    Unexpected behavior in rotation. Please file a bug report on github.
    Traceback (most recent call last):
      File "/home/runner/work/PyGLImER/PyGLImER/src/pyglimer/waveform/preprocessh5.py", line 346, in __station_process__
        st, _, infodict = __rotate_qc(
      File "/home/runner/work/PyGLImER/PyGLImER/src/pyglimer/waveform/preprocessh5.py", line 405, in __rotate_qc
        st.rotate(method='->ZNE', inventory=station_inv)
      File "/usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/obspy/core/stream.py", line 2733, in rotate
        return self._rotate_to_zne(inventory, **kwargs)
      File "/usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/obspy/core/stream.py", line 3469, in _rotate_to_zne
        self._rotate_specific_channels_to_zne(
      File "/usr/share/miniconda/envs/pyglimer/lib/python3.10/site-packages/obspy/core/stream.py", line 3542, in _rotate_specific_channels_to_zne
        raise NotImplementedError(msg)
    NotImplementedError: Unexpected behavior in rotation. Please file a bug report on github.

     30%|###       | 88/289 [00:04<00:14, 13.90it/s]
     31%|###1      | 90/289 [00:05<00:14, 13.57it/s]
     33%|###2      | 94/289 [00:05<00:12, 16.08it/s]
     37%|###6      | 106/289 [00:05<00:06, 30.27it/s]
     38%|###8      | 110/289 [00:05<00:07, 22.56it/s]
     40%|####      | 117/289 [00:05<00:06, 27.22it/s]
     42%|####1     | 121/289 [00:06<00:07, 21.04it/s]
     43%|####3     | 125/289 [00:06<00:07, 21.63it/s]
     44%|####4     | 128/289 [00:06<00:09, 16.91it/s]
     46%|####6     | 133/289 [00:06<00:07, 19.80it/s]
     65%|######4   | 187/289 [00:07<00:01, 90.08it/s]
     69%|######8   | 198/289 [00:07<00:01, 48.13it/s]
     71%|#######1  | 206/289 [00:08<00:02, 40.77it/s]
     74%|#######3  | 213/289 [00:08<00:01, 40.54it/s]
     76%|#######5  | 219/289 [00:08<00:02, 33.00it/s]
     83%|########2 | 239/289 [00:08<00:01, 48.47it/s]
     85%|########5 | 246/289 [00:09<00:00, 46.94it/s]
     87%|########7 | 252/289 [00:09<00:01, 34.57it/s]
     89%|########8 | 257/289 [00:09<00:01, 26.89it/s]
     91%|######### | 262/289 [00:09<00:01, 26.89it/s]
     92%|#########2| 266/289 [00:10<00:01, 15.64it/s]
     93%|#########3| 269/289 [00:10<00:01, 13.91it/s]
     94%|#########4| 272/289 [00:11<00:01, 14.32it/s]
     95%|#########4| 274/289 [00:11<00:01, 11.84it/s]
     96%|#########5| 276/289 [00:11<00:01, 11.76it/s]
     96%|#########6| 278/289 [00:11<00:00, 11.90it/s]
     97%|#########6| 280/289 [00:12<00:00, 11.75it/s]    100%|##########| 289/289 [00:12<00:00, 24.05it/s]
    /home/runner/work/PyGLImER/PyGLImER/src/pyglimer/database/rfh5.py:376: UserWarning: The header contains an item of type <class 'obspy.core.util.attribdict.AttribDict'>. Information            of this type cannot be written to an hdf5 file.
      warnings.warn(
    /home/runner/work/PyGLImER/PyGLImER/src/pyglimer/database/rfh5.py:376: UserWarning: The header contains an item of type <class 'obspy.core.inventory.response.Response'>. Information            of this type cannot be written to an hdf5 file.
      warnings.warn(
    /home/runner/work/PyGLImER/PyGLImER/src/pyglimer/database/rfh5.py:376: UserWarning: The header contains an item of type <class 'obspy.core.event.resourceid.ResourceIdentifier'>. Information            of this type cannot be written to an hdf5 file.
      warnings.warn(
    100%|##########| 1/1 [00:12<00:00, 12.23s/it]    100%|##########| 1/1 [00:12<00:00, 12.23s/it]




.. GENERATED FROM PYTHON SOURCE LINES 164-172

++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

Here, again we have a change between the ``SAC`` workflow and the ``HDF5``
workflow because the RFs are saved in form of
:class:`pyglimer.database.rfh5.RFDataBase`s. It is important to note that the
``RFDataBase`` is an abstraction of an ``HDF5`` file that contains RF specific 
Header info that cannot be saved in ASDF. So, we import the database class
and query for the P receiver functions we computed. and output a stream.

.. GENERATED FROM PYTHON SOURCE LINES 172-189

.. code-block:: default


    from pyglimer.database.rfh5 import RFDataBase

    network = request_dict['network']
    station = request_dict['station']

    path_to_rf = os.path.join(proj_dir, 'waveforms','RF','P', f'{network}.{station}.h5')

    # Open RF Database
    with RFDataBase(path_to_rf, 'r') as rfdb:

        # Query database for specific RFs
        rfstream = rfdb.get_data('IU', 'HRV', 'P', '*', 'rf')

    # Print number of RFs queried
    print(f"Number of RFs: {len(rfstream)}")





.. rst-class:: sphx-glr-script-out

 Out:

 .. code-block:: none

    Number of RFs: 7




.. GENERATED FROM PYTHON SOURCE LINES 190-194

PyGLImER is based on Obspy and hence, similar to ``obspy``, we can access
the ``Stats`` of a ``RFTrace`` in a ``RFStream``. The abstraction from obspy
follows a need for more (SAC) header information to handle RFs. Let's take
a look at the attributes.

.. GENERATED FROM PYTHON SOURCE LINES 194-201

.. code-block:: default


    from pprint import pprint

    rftrace = rfstream[0]
    pprint(rftrace.stats)






.. rst-class:: sphx-glr-script-out

 Out:

 .. code-block:: none

    Stats({'sampling_rate': 10.0, 'delta': 0.1, 'starttime': UTCDateTime(2021, 1, 10, 4, 4, 10, 294500), 'endtime': UTCDateTime(2021, 1, 10, 4, 6, 40, 194500), 'npts': 1500, 'calib': 1.0, 'network': 'IU', 'station': 'HRV', 'location': '60', 'channel': 'PRF', '_format': 'ASDF', 'back_azimuth': 175.0198780525632, 'distance': 66.41981345929631, 'event_depth': 217000.0, 'event_latitude': -24.0412, 'event_longitude': -66.5729, 'event_magnitude': 6.1, 'event_time': UTCDateTime(2021, 1, 10, 3, 54, 14, 483000), 'onset': UTCDateTime(2021, 1, 10, 4, 4, 40, 294000), 'phase': 'P', 'pol': 'v', 'processing': ["ObsPy 1.3.0: remove_response(fig=None::inventory=None::output='VEL'::plot=False::pre_filt=None::taper=True::taper_fraction=0.05::water_level=60::zero_mean=True)", "ObsPy 1.3.0: detrend(options={}::type='demean')", "ObsPy 1.3.0: taper(max_length=None::max_percentage=0.05::side='both'::type='hann')", 'ObsPy 1.3.0: trim(endtime=UTCDateTime(2021, 1, 10, 4, 6, 40, 194538)::fill_value=None::nearest_sample=True::pad=False::starttime=UTCDateTime(2021, 1, 10, 4, 4, 10, 294538))', 'ObsPy 1.3.0: normalize(norm=None)', "ObsPy 1.3.0: filter(options={'freq': 1.5, 'zerophase': True, 'corners': 2}::type='lowpass')"], 'slowness': 6.33734875354112, 'station_elevation': 200.0, 'station_latitude': 42.5064, 'station_longitude': -71.5583, 'type': 'time'})




.. GENERATED FROM PYTHON SOURCE LINES 202-218

First Receiver function plots
-----------------------------

If the Receiver functions haven't been further processed,
they are plotted as a function of time. A single receiver
function in the stream will be plotted as function of time
only. A full stream can make use of the distance measure saved
in the sac-header and plot an entire section as a function of
time and epicentral distance.

Plot single RF
++++++++++++++

Below we show how to plot the receiver function
as a function of time, and the clean option, which plots
the receiver function without any axes or text.

.. GENERATED FROM PYTHON SOURCE LINES 218-224

.. code-block:: default


    from pyglimer.plot.plot_utils import set_mpl_params

    # Plot RF
    rftrace.plot()




.. image-sg:: /tutorials/images/sphx_glr_DataCollectionHDF5_002.png
   :alt: DataCollectionHDF5
   :srcset: /tutorials/images/sphx_glr_DataCollectionHDF5_002.png
   :class: sphx-glr-single-img


.. rst-class:: sphx-glr-script-out

 Out:

 .. code-block:: none


    <AxesSubplot:xlabel='Conversion Time [s]', ylabel='A    '>



.. GENERATED FROM PYTHON SOURCE LINES 225-226

Let's zoom into the first 20 seconds (~200km)

.. GENERATED FROM PYTHON SOURCE LINES 226-229

.. code-block:: default


    rftrace.plot(lim=[0,20])




.. image-sg:: /tutorials/images/sphx_glr_DataCollectionHDF5_003.png
   :alt: DataCollectionHDF5
   :srcset: /tutorials/images/sphx_glr_DataCollectionHDF5_003.png
   :class: sphx-glr-single-img


.. rst-class:: sphx-glr-script-out

 Out:

 .. code-block:: none


    <AxesSubplot:xlabel='Conversion Time [s]', ylabel='A    '>



.. GENERATED FROM PYTHON SOURCE LINES 230-235

Plot RF section
+++++++++++++++

Since we have an entire stream of receiver functions at hand,
we can plot a section

.. GENERATED FROM PYTHON SOURCE LINES 235-238

.. code-block:: default


    rfstream.plot(scalingfactor=1)




.. image-sg:: /tutorials/images/sphx_glr_DataCollectionHDF5_004.png
   :alt: PRF component
   :srcset: /tutorials/images/sphx_glr_DataCollectionHDF5_004.png
   :class: sphx-glr-single-img


.. rst-class:: sphx-glr-script-out

 Out:

 .. code-block:: none


    <AxesSubplot:title={'center':'PRF component'}, xlabel='$\\Delta$ [$^{\\circ}$]', ylabel='Time [s]'>



.. GENERATED FROM PYTHON SOURCE LINES 239-241

Similar to the single RF plot we can provide time and 
epicentral distance limits:

.. GENERATED FROM PYTHON SOURCE LINES 241-248

.. code-block:: default


    timelimits = (0, 20)  # seconds  
    epilimits = (32, 36)  # epicentral distance
    rfstream.plot(
        scalingfactor=0.25, lim=timelimits, epilimits=epilimits,
        linewidth=0.75)




.. image-sg:: /tutorials/images/sphx_glr_DataCollectionHDF5_005.png
   :alt: PRF component
   :srcset: /tutorials/images/sphx_glr_DataCollectionHDF5_005.png
   :class: sphx-glr-single-img


.. rst-class:: sphx-glr-script-out

 Out:

 .. code-block:: none


    <AxesSubplot:title={'center':'PRF component'}, xlabel='$\\Delta$ [$^{\\circ}$]', ylabel='Time [s]'>



.. GENERATED FROM PYTHON SOURCE LINES 249-251

By increasing the scaling factor and removing the plotted lines, we can
already see trends:

.. GENERATED FROM PYTHON SOURCE LINES 251-257

.. code-block:: default


    rfstream.plot(
        scalingfactor=0.5, lim=timelimits, epilimits=epilimits, 
        line=False)





.. image-sg:: /tutorials/images/sphx_glr_DataCollectionHDF5_006.png
   :alt: PRF component
   :srcset: /tutorials/images/sphx_glr_DataCollectionHDF5_006.png
   :class: sphx-glr-single-img


.. rst-class:: sphx-glr-script-out

 Out:

 .. code-block:: none


    <AxesSubplot:title={'center':'PRF component'}, xlabel='$\\Delta$ [$^{\\circ}$]', ylabel='Time [s]'>



.. GENERATED FROM PYTHON SOURCE LINES 258-259

As simple as that you can create your own receiver functions with 
just a single smalle script.


.. rst-class:: sphx-glr-timing

   **Total running time of the script:** ( 1 minutes  5.448 seconds)


.. _sphx_glr_download_tutorials_DataCollectionHDF5.py:


.. only :: html

 .. container:: sphx-glr-footer
    :class: sphx-glr-footer-example



  .. container:: sphx-glr-download sphx-glr-download-python

     :download:`Download Python source code: DataCollectionHDF5.py <DataCollectionHDF5.py>`



  .. container:: sphx-glr-download sphx-glr-download-jupyter

     :download:`Download Jupyter notebook: DataCollectionHDF5.ipynb <DataCollectionHDF5.ipynb>`


.. only:: html

 .. rst-class:: sphx-glr-signature

    `Gallery generated by Sphinx-Gallery <https://sphinx-gallery.github.io>`_
